src/init/replace_vulnerability_detector.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+
+# Copyright (C) 2015, Wazuh Inc.
+# All rights reserved.
+# Wazuh.com
+
+# This program is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public
+# License (version 2) as published by the FSF - Free Software
+# Foundation.
+
+# Shell script that updates 'vulnerability-detector' to 'vulnerability-detection'
+
+isVulnerabilityDetectionInstalled()
+{
+    local OSSEC_CONFIGURATION_FILE="$1"
+    local VULNERABILITY_DETECTION_PATTERN="<vulnerability-detection>"
+
+    if ( grep -q "$VULNERABILITY_DETECTION_PATTERN" "$OSSEC_CONFIGURATION_FILE" ); then
+        return 0
+    fi
+
+    return 1
+}
+
+updateVulnerabilityDetector()
+{
+    if [ "$INSTYPE" = "server" ]; then
+        local OSSEC_CONFIGURATION_FILE="$1"
+
+        if isVulnerabilityDetectionInstalled "$OSSEC_CONFIGURATION_FILE"; then 
+            # Skip the update of vulnerability-detection if its already installed
+            return 0
+        fi
+
+        local OSSEC_CONFIGURATION_FILE_TMP="$1.tmp"
+        local VULN_TEMPLATE="./etc/templates/config/generic/wodle-vulnerability-detection.manager.template"
+        local INDEXER_TEMPLATE="./etc/templates/config/generic/wodle-indexer.manager.template"
+
+        touch $OSSEC_CONFIGURATION_FILE_TMP
+
+        local OSSEC_CONFIGURATION_FILE_BEFORE_VD="$(sed -ne '/<vulnerability-detector>/q;p' $OSSEC_CONFIGURATION_FILE)"
+        local OSSEC_CONFIGURATION_FILE_AFTER_VD="$(sed -e '1,/<\/vulnerability-detector>/d' $OSSEC_CONFIGURATION_FILE)"
```

Comment:
Tested an update from 4.7 to 4.8, and this logic breaks if there is no `<vulnerability-detector>` configuration block (which is a valid configuration)

```xml
</ossec_config>

  <vulnerability-detection>
    <enabled>yes</enabled>
    <index-status>yes</index-status>
    <feed-update-interval>60m</feed-update-interval>
  </vulnerability-detection>

  <indexer>
    <enabled>yes</enabled>
    <hosts>
      <host>https://0.0.0.0:9200</host>
    </hosts>
    <username>admin</username>
    <password>admin</password>
    <ssl>
      <certificate_authorities>
        <ca>/etc/filebeat/certs/root-ca.pem</ca>
      </certificate_authorities>
      <certificate>/etc/filebeat/certs/filebeat.pem</certificate>
      <key>/etc/filebeat/certs/filebeat-key.pem</key>
    </ssl>
  </indexer>
```

Reply:
fixed 👀

Reply:
Indeed!